toolchain/check/testdata/impl/assoc_const_self.carbon

```diff
@@ -57,6 +57,48 @@ impl () as Core.ImplicitAs(C) {
 // CHECK:STDERR:
 impl C as I where .V = () {}
 
+// --- fail_monomorphization_failure.carbon
+
+library "[[@TEST_NAME]]";
+
+interface I(N:! Core.IntLiteral()) {
+  // CHECK:STDERR: fail_monomorphization_failure.carbon:[[@LINE+3]]:12: error: array bound of -1 is negative [ArrayBoundNegative]
+  // CHECK:STDERR:   let V:! [Self; N];
+  // CHECK:STDERR:            ^~~~
```

Comment:
Unrelated to this PR, but this looks like it is highlighting the wrong part of the array type.